os/pm/pm_sleep.c

```diff
@@ -73,8 +73,11 @@
  * Public Functions
  ************************************************************************/
 
-static void pm_timer_callback(int argc, uint32_t sem)
+static void pm_timer_callback(int argc, uint32_t sem, uint32_t pid)
 {
+#ifdef CONFIG_PM_METRICS
+	pm_metrics_update_sleep((pid_t)pid);
```

Comment:
please find wd_setwakeupsource()

we can use timer as wakeup source using api rather than pm_sleep.
Could you consider how to add this?

Reply:
Currently whole `pm_metrics` logic is inside PM module, if we want to move logic into `wd_setwakeupsource` then we need to change pm interface and give `pm_metrics` API.